src/som/interpreter/bc/interpreter.py

```diff
@@ -32,8 +33,9 @@ def _do_super_send(bytecode_index, method, stack, stack_ptr):
     receiver = stack[stack_ptr - (num_args - 1)]
 
     if invokable:
+        first = method.get_inline_cache(bytecode_index)
```

Comment:
Is this really needed?

We're quickening the bytecode afterwards.
So, I would not expect there to be an inline cache entry.

For super, we are also guaranteed to have only a single target.
So, we don't really need an inline cache node here.

In the AST interpreter, we cache the method directly, here: https://github.com/SOM-st/PySOM/blob/master/src/som/interpreter/ast/nodes/message/super_node.py#L41-L44

Instead of setting the node here, we might perhaps just store the method?

Reply:
That was plumbing to adapt the old code, but there doesn't seem to be any need for inline caches there indeed. Sorry, but where/how would you store the method?

Reply:
Directly into the same array where you store the dispatch nodes otherwise?

Reply:
I thought it might be the case, I just didn't like that solution since I thought we may end up in a case where we expect a node but get a method instead. Another more important issue is that RPython doesn't seem to allow it, since the fact that the array can contain either a method or a node doesn't mesh well with its static analysis. I can create an entirely new field to hold the method in this case, I suppose...

Reply:
> may end up in a case where we expect a node but get a method instead
By construction, this should not be possible since the array indexes correspond to bytecodes and the bytecodes are either normal sends or super sends, but they can't be both.

If RPython can't be convinced in other ways, you may need to introduce a dispatch node for the desired behavior.

> I can create an entirely new field to hold the method in this case, I suppose...

I wouldn't want a mostly empty array, if that's what you mean with "new field".
It can't be a single field in the method object, since a method can have an arbitrary number of super sends.

Reply:
Couldn't find a way to convince RPython, so I stuck with the dispatch node...

Reply:
This lookup is removed now. Marking as resolved.